app/controllers/students_base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class StudentsBaseController < ApplicationController
+  before_action :require_student
+
+  def require_student
+    render file: 'public/404.html' unless current_student? 
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.<br>Trailing whitespace detected.